### PR TITLE
Rewrite deriche filter- add test - add python wrapper

### PR DESCRIPTION
**Pull request by LaurentBerger**
Use same notation as paper-
add test (using impulse repsonse) 
python warpper 
python example

### CODE REVIEW

**Comment by alalek** [comment]
Target branch is "master" or "3.4"?

**Comment by LaurentBerger** [comment]
3.4. I changed based branch : is it good?

**Comment by alalek** [comment]
- change "base" branch of this PR: master => 3.4 (use "Edit" button near PR title)
- rebase your commits from master onto 3.4 branch. For example:
    `git rebase -i --onto upstream/3.4 upstream/master`
    (check list of your commits, save and quit (Esc + "wq" + Enter)
    where `upstream` is configured by following [this GitHub guide](https://help.github.com/articles/configuring-a-remote-for-a-fork/) and fetched (`git fetch upstream`).
- push rebased commits into source branch of your fork (with `--force` option)

Note: no needs to re-open PR, apply changes "inplace".

File: modules/ximgproc/test/test_deriche_filter.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+#include "test_precomp.hpp"
+
+namespace opencv_test {
+    namespace {
+
+        TEST(ximgproc_DericheFilter, regression)
+        {
```
Comment (alalek) [inline modules/ximgproc/test/test_deriche_filter.cpp]: Please avoid unnecessary indentation in namespaces.

File: modules/ximgproc/test/test_deriche_filter.cpp [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            for (int i = 0; i < img.rows; i++)
+            {
+                double n = -31 + i;
+                for (int j = 0; j < img.cols; j++)
+                {
+                    double m = -31 + j;
+                    double x = -c * exp(-a * fabs(m))*sin(w*m);
+                    x = x * (k*(a*sin(w*fabs(n)) + w * cos(w*fabs(n)))*exp(-a * fabs(n))) / (a*a + w * w);
+                    x = x / (w*w);
+					float xx=static_cast<float>(x);
```
Comment (alalek) [inline modules/ximgproc/test/test_deriche_filter.cpp]: Please use 'spaces' only.

**Pull request merged**